detox/src/utils/environment.js

```diff
@@ -187,7 +187,31 @@ async function getFrameworkPath() {
 async function _doGetFrameworkPath() {
   const detoxVersion = getDetoxVersion();
   const sha1 = (await exec(`(echo "${detoxVersion}" && xcodebuild -version) | shasum | awk '{print $1}'`)).stdout.trim();
-  return `${DETOX_LIBRARY_ROOT_PATH}/ios/${sha1}/Detox.framework`;
+  return `${DETOX_LIBRARY_ROOT_PATH}/ios/framework/${sha1}/Detox.framework`;
+}
+
+let _iosXCUITestRunnerPath;
+
+async function getXCUITestRunnerPath() {
+  if (!_iosXCUITestRunnerPath) {
+    _iosXCUITestRunnerPath = _doGetXCUITestRunnerPath();
+  }
+
+  return _iosXCUITestRunnerPath;
+}
+
+async function _doGetXCUITestRunnerPath() {
+  const detoxVersion = getDetoxVersion();
+  const sha1 = (await exec(`(echo "${detoxVersion}" && xcodebuild -version) | shasum | awk '{print $1}'`)).stdout.trim();
```

Comment:
1. I don't like the fact we're calling a pretty complex command twice. This could have been memoized.
2. Obviously, this is something that requires a separate function.
3. It's quite trivial to calculate a sha1 sum in Node.js itself (in case of doubt, ask any AI assistant), so maybe we could get away with just calling `xcodebuild -version` in Shell? This should be perfectly enough, no need to `echo`, `awk` and `shasum` it, right?

Reply:
1+2. It's already memoized
https://github.com/wix/Detox/blob/7bfba227375ab4241a66c08c5f6895d45c267eb9/detox/src/utils/environment.js#L193-L201

Reply:
3. you're right.

Reply:
done

Reply:
Same :)